hw/mcu/nordic/nrf5340_net/src/hal_spi.c

```diff
@@ -685,10 +686,14 @@ hal_spi_txrx_noblock(int spi_num, void *txbuf, void *rxbuf, int len)
 {
     struct nrf5340_net_hal_spi *spi = &nrf5340_net_hal_spi0;
 
-    if (spi_num != 0 || (spi->txrx_cb_func == NULL) || (len == 0)) {
+    if (spi_num != 0 || (spi->txrx_cb_func == NULL) || (len == 0) || !nrfx_is_in_ram(txbuf)) {
         return EINVAL;
     }
 
+    if (rxbuf != NULL && !nrfx_is_in_ram(rxbuf)) {
+        return SYS_EINVAL;
```

Comment:
This looks slightly strange to return SYS_EINVAL and EINVAL so close together for similar reason.
Maybe some unification could be applied, I'm not sure which way though.

Reply:
ah yes, let me be consistent with the existing convention.

Reply:
addressed